_playwright-tests/test-utils/src/helpers/poll.ts

```diff
@@ -16,3 +24,23 @@ export const poll = async (
 
 // sleep in ms, must await e.g. await sleep(num)
 export const sleep = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+/**
+ * Waits for a repository introspection to complete by polling the repository status.
+ * @param client - Configuration object for API client (optional, uses default if not provided)
+ * @param repoUuid - UUID of the repository to check
+ * @param interval - Polling interval in seconds (default: 10)
+ */
+export const waitForRepositoryIntrospection = async (
```

Comment:
the name here is kinda incorrect, it doesn't wait for only the introspection, but while the status is pending, that can also mean snapshoting, so `waitWhileRepositoryIsPending` would be a better name 😅